### PR TITLE
Initial e2e test suite

### PR DESCRIPTION
**Pull request by jakolehm**
Currently has setup/teardown only for vagrant.

```
$ rake vagrant:setup
$ bundle exec rspec -f d spec/
$ rake vagrant:teardown
```

### CODE REVIEW

**Comment by matti** [comment]
What's the preferred method to use install CLI with this?

**Comment by matti** [comment]
works.

File: test/spec/features/app/deploy_spec.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+require 'spec_helper'
+
+describe 'app deploy' do
+  it 'deploys a simple stack' do
```
Comment (jnummelin) [inline test/spec/features/app/deploy_spec.rb]: Seems to really deploy an app... :)

File: test/spec/features/app/ps_spec.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+require 'spec_helper'
+
+describe 'app ps' do
```
Comment (jnummelin) [inline test/spec/features/app/ps_spec.rb]: There's no guarantee that there is anything deployed, does it make sense to check the `app ps` without anything deployed? (IMO no)

File: test/spec/features/app/remove_spec.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
+require 'spec_helper'
+
+describe 'app remove' do
+  it 'removes a deployed stack' do
```
Comment (jnummelin) [inline test/spec/features/app/remove_spec.rb]: Removes an app... :)

File: test/spec/features/plugin/uninstall_spec.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+require 'spec_helper'
+
+describe 'plugin uninstall' do
+  before(:each) do
+    run('kontena plugin install aws')
+  end
+
+  it 'removes installed plugin' do
+    k = run('kontena plugin uninstall --force aws')
+    expect(k.code).to eq(0)
```
Comment (jnummelin) [inline test/spec/features/plugin/uninstall_spec.rb]: Should we also verfiy that we get at least some of the known plugins? (DO, AWS, vagrant, ...)

File: test/spec/features/service/deploy_spec.rb [hunk 5]
```diff
@@ -0,0 +1,11 @@
+require 'spec_helper'
+
+describe 'service deploy' do
+  it 'deploys a service' do
+    run("kontena service create test-1 redis:3.0")
```
Comment (jnummelin) [inline test/spec/features/service/deploy_spec.rb]: Why service is created with `run(...)` and other command with `kommando`? What's the difference?

File: test/spec/features/service/scale_spec.rb [hunk 6]
```diff
@@ -0,0 +1,13 @@
+require 'spec_helper'
+
+describe 'service scale' do
+  it 'scales a service' do
+    run("kontena service create test-1 redis:3.0")
+    k = kommando("kontena service deploy test-1")
+    expect(k.run).to be_truthy
+    k = kommando("kontena service scale test-1 3")
+    expect(k.run).to be_truthy
+
```
Comment (jnummelin) [inline test/spec/features/service/scale_spec.rb]: Should we verify that three instances are actually created within some decent time?

File: test/spec/features/app/ps_spec.rb (hunk 2 above)
Comment (jakolehm) [inline test/spec/features/app/ps_spec.rb]: we should test both.. with and without deploy ;)

File: test/spec/features/service/scale_spec.rb (hunk 6 above)
Comment (jakolehm) [inline test/spec/features/service/scale_spec.rb]: Probably yes, although scale blocks until deployment has succeeded.

File: test/spec/features/plugin/uninstall_spec.rb (hunk 4 above)
Comment (jakolehm) [inline test/spec/features/plugin/uninstall_spec.rb]: I don't get this? Can you explain a bit more.

File: test/spec/features/service/deploy_spec.rb (hunk 5 above)
Comment (jakolehm) [inline test/spec/features/service/deploy_spec.rb]: `support/shell.rb` should explain.

File: test/spec/features/plugin/uninstall_spec.rb (hunk 4 above)
Comment (jnummelin) [inline test/spec/features/plugin/uninstall_spec.rb]: Ignore, was reading badly :)

**Pull request merged**